### ISSUE TITLE
New rule: avoid-prefixing-tests-with

### ISSUE THREAD

**Issue by G-Rath** [jest-community/eslint-plugin-jest#273]
Basically, this would catch things like `it('it should throw an error', () => { ...`

I figure it would be best as a more generic rule, that took a list of strings that should be considered as undesired to have at the start of the messages for test calls, with "it" being the default.

(The rule name could be improved :/)

**Comment by ranyitz** [comment]
It would also help if you want to remove an accidental space:

```js
it(' should remove space in the beginning', () => { ...
```

I think that adding a rule like that which is fixable (removes the unwanted string) would be nice.

**Comment by SimenB** [comment]
Sure, PR welcome! 👍

**Comment by suganya-sangith** [comment]
Shall I start working on it?

**Comment by G-Rath** [comment]
@suganya-sangith if you've got the time, go for it! I'd love to see this rule, but haven't had the time to finish off my implementation.

Feel free to tag me in the PR, and I'll review it 🙂
